lib/zhr_devs/bakery_integration/command_runner.ex

```diff
@@ -1,53 +1,59 @@
-# defmodule ZhrDevs.BakeryIntegration.CommandRunner do
-#   @moduledoc """
-#   Generic process that will run the given command
-#   """
-#   use GenServer
+defmodule ZhrDevs.BakeryIntegration.CommandRunner do
+  @moduledoc """
+  Generic process that will run the given command
+  """
+  use GenServer
 
-#   alias Uptight.Text, as: T
+  alias Uptight.Text, as: T
 
-#   import Witchcraft.Functor
+  import Witchcraft.Functor
 
-#   require Logger
+  require Logger
 
-#   defmodule State do
-#     defstruct [:port, :latest_output, :exit_status]
-#   end
+  defmodule State do
+    defstruct [:port, :latest_output, :exit_status]
+  end
 
-#   def init(%Ubuntu.Command{} = cmd) do
-#     port =
-#       Port.open(
-#         {:spawn_executable, cmd.path |> Ubuntu.Path.render() |> T.un()},
-#         [:binary, args: cmd.args |> map(&T.un/1)]
-#       )
+  def start_link(opts \\ []) do
+    GenServer.start_link(__MODULE__, opts)
+  end
 
-#     Port.monitor(port)
+  def init(opts) do
+    cmd = Keyword.fetch!(opts, :cmd)
 
-#     {:ok, %State{port: port}}
-#   end
+    port =
+      Port.open(
+        {:spawn_executable, cmd.path |> Ubuntu.Path.render() |> T.un()},
+        [:binary, args: cmd.args |> map(&T.un/1)]
+      )
 
-#   # This callback handles data incoming from the command's STDOUT
-#   def handle_info({port, {:data, text_line}}, %{port: port} = state) do
-#     Logger.info "Data: #{inspect text_line}"
-#     {:noreply, %State{state | latest_output: String.trim(text_line)}}
-#   end
+    Port.monitor(port)
```

Comment:
What does this function do? @maslowalex

Reply:
https://hexdocs.pm/elixir/1.12/Port.html#monitor/1

Reply:
It acts like a supervisor, once port will exit - we will receive the message about the how it terminated.